[CloudFront] list_distributions doesn't properly return distribution_config

Attached test shows that the Cloudfront distribution_config fields aren't being returned as top level fields as mentioned in 
https://docs.aws.amazon.com/boto3/latest/reference/services/cloudfront/client/list_distributions.html

```
(.venv) jlim@603e5f5a8790 moto % pytest tests/test_cloudfront/test_cloudfront_distributions.py::test_list_distributions_config_fields -s 
=============================================================================================================================================================================================== test session starts ================================================================================================================================================================================================
platform darwin -- Python 3.11.4, pytest-8.3.4, pluggy-1.5.0
rootdir: /Users/iwt148/moto
configfile: pyproject.toml
plugins: cov-6.0.0, order-1.3.0, xdist-3.6.1
collected 1 item                                                                                                                                                                                                                                                                                                                                                                                                   

tests/test_cloudfront/test_cloudfront_distributions.py F

===================================================================================================================================================================================================== FAILURES =====================================================================================================================================================================================================
______________________________________________________________________________________________________________________________________________________________________________________ test_list_distributions_config_fields _______________________________________________________________________________________________________________________________________________________________________________________

    @mock_aws
    def test_list_distributions_config_fields():
        client = boto3.client("cloudfront", region_name="us-east-1")
    
        client.create_distribution(
            DistributionConfig=scaffold.example_distribution_config(ref="ref1")
        )
    
        item = client.list_distributions()["DistributionList"]["Items"][0]
>       assert item["Origins"]["Quantity"] == 1
E       KeyError: 'Origins'

tests/test_cloudfront/test_cloudfront_distributions.py:786: KeyError
============================================================================================================================================================================================= short test summary info ==============================================================================================================================================================================================
FAILED tests/test_cloudfront/test_cloudfront_distributions.py::test_list_distributions_config_fields - KeyError: 'Origins'
```

--- a/moto/cloudfront/models.py
+++ b/moto/cloudfront/models.py
@@ -237,6 +237,24 @@ class DistributionConfig:
         self.caller_reference = config["CallerReference"]
 
 
+DISTRIBUTION_CONFIG_FIELDS = [
+    "Aliases",
+    "Origins",
+    "OriginGroups",
+    "DefaultCacheBehavior",
+    "CacheBehaviors",
+    "CustomErrorResponses",
+    "Comment",
+    "PriceClass",
+    "Enabled",
+    "ViewerCertificate",
+    "Restrictions",
+    "WebACLId",
+    "HttpVersion",
+    "IsIPV6Enabled",
+]
+
+
 class Distribution(BaseModel, ManagedState):
     def __init__(self, account_id: str, region_name: str, config: dict[str, Any]):
         # Configured ManagedState
@@ -261,6 +279,11 @@ class Distribution(BaseModel, ManagedState):
     @property
     def location(self) -> str:
         return f"https://cloudfront.amazonaws.com/2020-05-31/distribution/{self.distribution_id}"
+
+    def __getattr__(self, name: str) -> Any:
+        if name in DISTRIBUTION_CONFIG_FIELDS:
+            return getattr(self.distribution_config, name)
+        raise AttributeError(name)
 
 
 class OriginAccessControl(BaseModel):

--- a/tests/test_cloudfront/test_cloudfront_distributions.py
+++ b/tests/test_cloudfront/test_cloudfront_distributions.py
@@ -784,7 +784,9 @@ def test_list_distributions_config_fields():
 
     item = client.list_distributions()["DistributionList"]["Items"][0]
     assert item["Origins"]["Quantity"] == 1
-    assert item["Origins"]["Items"][0]["DomainName"] == "asdf.s3.us-east-1.amazonaws.com"
+    assert (
+        item["Origins"]["Items"][0]["DomainName"] == "asdf.s3.us-east-1.amazonaws.com"
+    )
     assert item["DefaultCacheBehavior"]["ViewerProtocolPolicy"] == "allow-all"
     assert item["ViewerCertificate"]["CloudFrontDefaultCertificate"] is True
     assert item["Enabled"] is False

--- a/tests/test_cloudfront/test_cloudfront_distributions.py
+++ b/tests/test_cloudfront/test_cloudfront_distributions.py
@@ -772,3 +772,20 @@ def test_add_new_gateway_basic():
         result["CacheBehaviors"]["Items"][0]["OriginRequestPolicyId"]
         == new_behaviors[0]["OriginRequestPolicyId"]
     )
+
+
+@mock_aws
+def test_list_distributions_config_fields():
+    client = boto3.client("cloudfront", region_name="us-east-1")
+
+    client.create_distribution(
+        DistributionConfig=scaffold.example_distribution_config(ref="ref1")
+    )
+
+    item = client.list_distributions()["DistributionList"]["Items"][0]
+    assert item["Origins"]["Quantity"] == 1
+    assert item["Origins"]["Items"][0]["DomainName"] == "asdf.s3.us-east-1.amazonaws.com"
+    assert item["DefaultCacheBehavior"]["ViewerProtocolPolicy"] == "allow-all"
+    assert item["ViewerCertificate"]["CloudFrontDefaultCertificate"] is True
+    assert item["Enabled"] is False
+    assert item["Comment"] == "an optional comment that's not actually optional"

